frontend/packages/shared/src/components/altinnHeaderMenu/AltinnHeaderMenu.tsx

```diff
@@ -13,21 +15,30 @@ export interface AltinnHeaderMenuItem {
 }
 
 export const AltinnHeaderMenu = ({ activeSubHeaderSelection, menu }: IAltinnHeaderMenuProps) => {
-  if (!menu?.length) return null;
-
-  return (
-    <ul className={classes.menu} data-testid='altinn-header-menu'>
-      {menu.map((item) => (
-        <li
-          key={item.key}
-          className={classNames(
-            classes.menuItem,
-            activeSubHeaderSelection === item.key && classes.active
-          )}
-        >
-          {item.link}
-        </li>
-      ))}
-    </ul>
-  );
+  const { t } = useTranslation();
+  if (window.location.pathname.includes('preview')) {
+    return (
+      <Link to={''} className={classes.previewLinkElement}>
+        {t('general.preview')}
```

Comment:
This should be passed via the AltinnHeader component via the `menu` prop, not defined explicitly here. AltinnHeader and its sub-components should not be aware of the context in which they are used. The changes made to this file can be reverted.